updated to add access function role (#909)
couchbaselabs/docs-sync-gateway: modules/ROOT/pages/_partials/sync-api/sync-function-api-access.adoc

TIP: Prefix the `username` argument value with `role:` to apply this function to a role rather than a user.
This grants access to the specified channel(s) for all users assigned that role.

The effects of all access calls by all active documents are effectively combined in a union, so if any document grants a user access to a channel, that user has access to the channel.

> The `access()` function grants access to users or roles that do not yet exist.

Reviewer: ```suggestion
The `access()` function can grant access to users or roles that do not yet exist.
```

I don't know if this is better, but it could be that the users/roles already exist, or they do not exist.

> The effects of all access calls by all active documents are effectively combined in a union, so if _any_ document grants a user access to a channel, that user has access to the channel.

Reviewer 2: ```suggestion
The effects of all access calls by all active documents are effectively combined in a union, so if any document grants a user access to a channel, that user has access to the channel.
```
No italics, according to style guide :)

The `access()` function grants access to users or roles that do not yet exist.

> When you create such users or define roles later, the previous granted access permissions apply to them.

Reviewer 2: ```suggestion
When you create new users or define roles later, the previous granted access permissions apply to them.
```
